app/src/main/kotlin/com/example/android/picover/MainActivity.kt

```diff
@@ -32,18 +38,25 @@ class MainActivity : ComponentActivity() {
 					NavigationItem.Camera,
 					NavigationItem.Profile
 				)
-
 				Scaffold {
 					AnimatedVisibility(
 						visible = navigationType == NavigationType.NAVIGATION_BAR
 					) {
+						val coroutineScope = rememberCoroutineScope()
```

Comment:
i am not a compose expert, and i am not sure how to handle with the coroutine scope across the nested elements, for example, if passing is it is a good practice ?

Reply:
As far as I know, this is exactly the way how you should use it. Create a coroutine scope bound to the `AnimatedVisibility` composable lifecycle and create a new coroutine in its scope outside of a composable (e.g. in the `onClick()` lambda), so that it will be automatically canceled once it leaves the composition. To sum up `rememberCoroutineScope()` is designed to handle coroutine scope across nested elements. In another case, if you want to handle suspend functions inside a composable you should use `LaunchEffect`.